lib/rubocop/cop/rspec/nested_groups.rb

```diff
@@ -97,30 +97,34 @@ class NestedGroups < Cop
           "Configuration key `#{DEPRECATED_MAX_KEY}` for #{cop_name} is " \
           'deprecated in favor of `Max`. Please use that instead.'
 
-        def_node_search :find_contexts, ExampleGroups::ALL.block_pattern
+        def_node_search :find_example_groups, ExampleGroups::ALL.block_pattern
+        def_node_matcher :example_group?, ExampleGroups::ALL.block_pattern
 
         def on_top_level_describe(node, _args)
-          find_nested_contexts(node.parent) do |context, nesting|
+          find_nested_example_groups(node.parent) do |example_group, nesting|
             self.max = nesting
             add_offense(
-              context.send_node,
+              example_group.send_node,
               message: message(nesting)
             )
           end
         end
 
         private
 
-        def find_nested_contexts(node, nesting: 1, &block)
-          find_contexts(node) do |nested_context|
-            yield(nested_context, nesting) if nesting > max_nesting
+        def find_nested_example_groups(node)
+          find_example_groups(node) do |example_group|
+            nesting = nesting_count(example_group)
```

Comment:
Do you need to count upwards or we can keep the count while going down the tree?
It's an open question - the code is fine and I guess there is no much performance difference, but still, seem like recursive search + count is the standard way to achieve this. Maybe just not feasible with the new implementation?

Reply:
There is an issue with tracking counter manually with recursion. There may be user defined blocks which should be ignored:

```ruby
describe 'foo' do
  [].each do |i|
    context 'bar' do
    end
  end
end
```

Actually solution with tracking shouldn't be too tricky

Reply:
If it makes code more complicated, let's keep it this way

Reply:
Checked this approach. Seems pretty simple.